ddtrace/tracer/sampler.go

```diff
@@ -147,3 +154,225 @@ func (ps *prioritySampler) apply(spn *span) {
 	}
 	spn.SetTag(keySamplingPriorityRate, rate)
 }
+
+type rulesSampler struct {
```

Comment:
Can we describe in detail how the rule sampler works? Explain rules, rate limiter, etc.

Reply:
Added a big comment.

Reply:
Thanks Caleb. That's great.